Sources/FioriARKit/BarcodeScanning/Views/ScanningSplitView.swift

```diff
@@ -107,49 +108,49 @@ struct AddBarcodeSheet: View {
     @Binding var neededBarcodes: [BarcodeModel]
     @Binding var isPresented: Bool
     
-    @State var payload = ""
+    @State var currentPayload: BarcodeModel = .empty
+    @State var needBarcodes: [BarcodeModel] = []
+    @State var foundPayloads: Set<BarcodeModel> = []
     @State var title = ""
-    
     @State var added: Bool = false
     
     var body: some View {
         VStack {
             HStack {
-                Text("Add Barcode")
-                    .font(.system(size: 22))
-                    .foregroundColor(.white)
-                Spacer()
                 Button("Dismiss") {
                     isPresented.toggle()
-                }.font(.system(size: 17))
-            }
-            
-            TextField("Payload...", text: $payload)
-                .textFieldStyle(PlainTextFieldStyle())
-                .padding(.bottom, 10)
-            TextField("Title...", text: $title)
-                .textFieldStyle(PlainTextFieldStyle())
-            
-            Button(action: {
-                neededBarcodes.append(BarcodeModel(id: payload, title: title, isDiscovered: false, symbology: .ean13)) // TODO: Have user select symbology
-                title = ""
-                payload = ""
-                added = true
-                DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
-                    added = false
                 }
-            }, label: {
-                Text(added ? "Success!" : "Add")
-                    .frame(width: 200, height: 40)
-                    .foregroundColor(.white)
-                    .background(
-                        RoundedRectangle(cornerRadius: 8)
-                            .fill(Color.preferredColor(.tintColor, background: .lightConstant))
-                    )
-            })
+                Spacer()
+                Button(action: {
+                    neededBarcodes.append(BarcodeModel(id: currentPayload.id, title: title, isDiscovered: false, symbology: .ean13)) // TODO: Have user select symbology
+                    title = ""
+                    currentPayload = .empty
+                    added = true
+                    DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
+                        added = false
+                    }
+                }, label: {
+                    Text(added ? "Success!" : "Add Barcode")
+                })
+            }.padding()
             
-            Spacer()
+            VStack{
+                HStack{
+                    Text("Title:")
+                    TextField("...", text: $title)
+                        .textFieldStyle(PlainTextFieldStyle())
+                }
+                HStack{
+                    Text("Payload:")
+                    Text(currentPayload.id).frame(
```

Comment:
Overall looks good to me.

Just add a line break so the .frame(maxWidth: .infinity, alignment: .leading) its on the next line and all on one line. Just a SwiftUI convention for view modifiers.